src/lib.rs

```diff
@@ -98,6 +98,8 @@ impl<T> Gc<T> {
         unsafe {
             let baseptr = GC_ALLOCATOR.alloc(layout).unwrap().as_ptr();
             let objptr = baseptr.add(uoff);
+
+            AllocMetadata::insert(objptr as usize, layout.size(), true);
```

Comment:
I see, so you do the insertion here...

Reply:
Yes, and it's subtly different. We insert the `objptr` for GC objects, not the `baseptr`. This means that theoretically, if there existed only a pointer on the stack to a GC object's `baseptr`, it would not be considered reachable and would be reclaimed.